### PR TITLE
Arrow list

### PR DESCRIPTION
**Pull request by haixuanTao**
This is a PR is a follow up PR to #6. It adds:
- all basic type that were missing 
- It links `DataType::List` Type to ROS2 array, sequence and bounded sequence. Some further optimization can potentially be done such as recognizing a list of basic type (u8, i16...) and applying a buffer copy.

## Notes

Some parts are definitely a bit verbose and could definitely be refactored in nicer ways.

### CODE REVIEW

File: python/src/typed/mod.rs [hunk 1]
```diff
@@ -60,63 +64,88 @@ fn type_info_for_member(
...
+            BasicType::I64 => DataType::Int64,
+            BasicType::U8 => DataType::UInt8,
+            BasicType::U16 => DataType::UInt16,
+            BasicType::U32 => DataType::UInt32,
+            BasicType::U64 => DataType::UInt64,
+            BasicType::F32 => DataType::Float32,
+            BasicType::F64 => DataType::Float64,
+            BasicType::Bool => DataType::Boolean,
+            BasicType::Char => DataType::Utf8,
+            BasicType::Byte => DataType::UInt8,
```
Comment (phil-opp) [inline python/src/typed/mod.rs:126]: Is `type_info_for_member` and `type_info_for_nestable_type` even used anymore?

File: python/src/typed/mod.rs [hunk 2]
```diff
@@ -126,36 +155,130 @@ pub fn default_for_member(
...
+                default_for_nestable_type(t, package_name, messages)?
+            }
+        },
+        MemberType::Array(array) => match &m.default.as_deref() {
+            Some([]) => eyre::bail!("empty default value not supported"),
+            Some([default]) => preset_default_for_basic_type(&array.value_type, &default)
+                .with_context(|| format!("failed to parse default value for `{}`", m.name))?,
+            Some(_) => {
+                eyre::bail!("there should be only a single default value for non-sequence types")
+            }
```
Comment (phil-opp) [inline python/src/typed/mod.rs:177]: Why are empty arrays and arrays with multiple entries invalid as default values?

File: python/src/typed/mod.rs (hunk 1 above)
Comment (haixuanTao) [reply python/src/typed/mod.rs:126]: It's not, thanks!

File: python/src/typed/mod.rs (hunk 2 above)
Comment (haixuanTao) [reply python/src/typed/mod.rs]: I have fixed this issue in e55be5c79d431bb4d9fb2ca46a8f7480677a6bad

**Review by phil-opp — APPROVED** [review]

**Comment by phil-opp** [comment]
Thanks!

**Pull request merged**